lib/perl/Genome/InstrumentData/AlignmentResult.pm

```diff
@@ -1146,6 +1146,22 @@ sub create_bam_header {
     return 1;
 }
 
+sub set_bam_size {
+    my ($self, $bam_file) = @_;
+    return 1 if $self->bam_size;
+
+    unless (defined $bam_file && -s $bam_file) {
+        $bam_file = $self->get_bam_file;
+        unless (-s $bam_file) {
+            die $self->error_message('BAM file (%s) does not exist or is empty', $bam_file);
+        }
+    }
+    return 1 if $self->bam_size;
```

Comment:
If the idea here is to check if another process has set `bam_size` in the meantime, then we'll need to do something to ensure this checks the database again.  (For example, setting our `UR::Context`'s `query_underlying_context` property to `1` temporarily.)